src/components/Application/ProgressBar.tsx

```diff
@@ -68,95 +79,205 @@ const ProgressBar: React.FC<ProgressBarProps> = ({
+    const isStarted = progress.doneNonRequired + progress.doneRequired > 0
+    const typeIndex = type === ProgressType.section ? 0 : 1
+
+    if (isStrictlylInvalid) return progressIconMap.error[typeIndex]
+    if (completed && valid) return progressIconMap.completed[typeIndex]
+    if (isActive) return progressIconMap.current[typeIndex]
+    if (isStarted) return progressIconMap.incomplete[typeIndex]
+    return progressIconMap.notStarted[typeIndex]
+  }
+
   const getPageList = (
     sectionCode: string,
     pages: { [pageNumber: string]: Page },
-    isStrictSection: boolean
+    isStrictSection: boolean,
+    sectionProgress: ApplicationProgress
   ) => {
     const isActivePage = (sectionCode: string, pageNumber: number) =>
       currentSectionCode === sectionCode && Number(page) === pageNumber
 
     const checkPageIsStrict = (pageNumber: string) =>
       isStrictSection && strictSectionPage?.pageNumber === Number(pageNumber)
 
-    return (
-      <List
-        link
-        style={inlineStyles.link}
-        items={Object.entries(pages).map(([number, { name: pageName, progress }]) => ({
-          key: `ProgressSection_${sectionCode}_${number}`,
-          active: isActivePage(sectionCode, Number(number)),
-          as: 'a',
-          icon: progress ? getIndicator(progress, checkPageIsStrict(number)) : null,
-          content: pageName,
-          onClick: () => handleChangeToPage(sectionCode, Number(number)),
-        }))}
-      />
-    )
-  }
+    const SectionProgressBar: React.FC<ProgressBarProps> = ({ percent, length, error }) => {
+      return (
+        <Progress
+          percent={percent}
+          size="tiny"
+          success={!error}
+          error={error}
+          style={{
+            width: length,
+            transformOrigin: '0 0',
+            transform: `rotate(90deg) translate(2px, -10px)`,
+          }}
+        />
+      )
+    }
 
-  // We want to show three states:
-  // error -> if at least one error or if not completed and strict
-  // success -> if completed and valid
-  // empty circle with number -> if none of the above and section (has step), also add key
-  // or empty circle -> if none of the above
-  const getIndicator = (progress: ApplicationProgress, isStrict: boolean, step?: number) => {
-    const { completed, valid } = progress
-    const isStrictlylInvalid = !valid || (isStrict && !completed)
-    const size = step ? 'large' : 'small'
+    const calculateBarPercent = () =>
+      sectionProgress.totalRequired === 0
+        ? 100
+        : (sectionProgress.doneRequired / sectionProgress.totalRequired) * 100
 
-    if (isStrictlylInvalid) return <Icon name={'exclamation circle'} color={'red'} size={size} />
-    if (completed && valid) return <Icon name={'check circle'} color={'green'} size={size} />
+    const calculateBarLength = () => {
+      // Adjust these constants to match existing style settings
+      const pixelsPerPage = 27.8
```

Comment:
I am ok with this a little hacky compromise to make it look awesome, it's well contained, only improvement would be to move those to constants at the top of file

Reply:
Like this? 4452d378